File: 구현/11723_집합/11723_집합.js
let [n, ...commands] = `26
add 1
add 2
check 1
check 2
check 3
remove 2
check 1
check 2
toggle 3
check 1
check 2
check 3
check 4
all
check 10
check 20
toggle 10
remove 20
check 10
check 20
empty
check 1
toggle 1
check 1
toggle 1
check 1`.split('\n');

let set = new Set();

for (let i=0; i<n; i++) {
  let [command, val] = commands[i].split(' ');
  switch (command) {
    case 'add':
      set.add(+val);
      break;
    case 'remove':
      set.delete(+val)
      break;
    case 'check':
      set.has(+val)? console.log(1) : console.log(0);
      break;
    case 'toggle':
      set.has(+val)? set.delete(+val) : set.add(+val);
      break;
    case 'all':
      for (let i=1; i<21; i++) set.add(i);
      break;
    case 'empty':
      set.clear();
      break;
  }
}

File: 구현/11723_집합/11723_집합.py
import sys
input = sys.stdin.readline

# s = set()
# T = int(input())
# for tc in range(T) :
#     line = input().split()
#     if line[0] == "add":
#         s.add(int(line[1]))
#     elif line[0] == "remove":
#         s.discard(int(line[1]))
#     elif line[0] == "check":
#         if int(line[1]) in s:
#             print(1)
#         else:
#             print(0)
#     elif line[0] == "toggle":
#         if int(line[1]) in s:
#             s.discard(int(line[1]))
#         else:
#             s.add(int(line[1]))
#     elif line[0] == "all":
#         for i in range(1, 21):
#             s.add(i)
#     elif line[0] == "empty":
#         s.clear()

# 문제 속도가 빠른 사람의 풀이
s = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
for _ in range(int(input())):
    l = input().split()
    if l[0] == "add":
        s[int(l[1])] = 1
    elif l[0] == "remove":
        s[int(l[1])] = 0
    elif l[0] == "check":
        sys.stdout.write('1\n' if s[int(l[1])] else '0\n')
    elif l[0] == "toggle":
        s[int(l[1])] = 1 - s[int(l[1])]
    elif l[0] == "all":
        s = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
    elif l[0] == "empty":
        s = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]

